### ISSUE TITLE
Modern user interface

### ISSUE THREAD

**Issue by PancakeTAS** [MinecraftTAS/LoTAS#125]
New OpenGL rendered GUI for LoTAS 3.0
## Limitations
While everything is possible due to having OpenGL access, I wanna make this as simple as possible so I'm putting up these loose restrictions:
- Only cubes, no rounded edge or spheres
- No super fancy drop shadows or glows (simple drop shadows that are just 1 pixel offset are fine)
- Text must be Minecraft Font on font sizes 10, 20, 30 or any multiple of 10x (Canvas is 1080p, turn off font interpolation)
- Hovers, Animations and Gradients are allowed
## Goal
The goal of the GUI is to give the user access to all LoTAS features using a handy overlay. The screen can be opened anywhere, in-game, escape menu, *main menu*. The menu is split into following parts:
- Left and right should be filled with easy to access, quick, tools.
- Bottom should have buttons to open extra settings for a mod, or launch more complicated mods (such as spawn manipulation)
- The remaining canvas (center until top) should be the big room for the "extra settings". For example this could be used for entity manipulation.
The menu has to be expandable, meaning if we add future mods in the future, it won't be a problem and we can easily shift one or two elements to make room. The mods currently available are these:
- Tickrate Changer. For this mod I recommend a panel on the left with buttons for increasing, decreasing and freezing the tickrate and also shortcut buttons that put the player at customizable tickrates (simple as like a button with a number). Additionally a button that enables automatic tickrate zero on gui open would also be appreciated
- Dupe Mod. This mod is still very simple having only a panel on the left, with saving and loading buttons. No idea what to add to this
- Savestate Mod. This mod should have an extended menu at the bottom, as well as a quick access on the left for quick saving and loading. The extended menu should have a list of states, and next to it a name, description, date and preview screenshot of it (either when clicking on it or directly in the list). Buttons for loading and deleting should also be there.
- Dragon Manipulation. For now this mod should have a menu only visible if in the end, on the right, for changing the ender dragon into one of 4 modes: Off, Perch, Strafing, Holding Pattern. (names are subject to change)
In the future I would like to have these 2 mods as well:
- OBS Control. This mod connects to OBS websocket and sets up an NDI source. In LoTAS itself there should simply be a button for starting, stopping the recording and a checkmark/toggle for automatically splitting the recording on loadstate and savestate on the right of LoTAS.
- Loot Manipulation. Loot Tables are automatically read and parsed for blocks, entities and also misc stuff like explosions should be accessible *using an extended menu*. Also while we're add it let's add the direction of item drops in here as well.

Mods that are removed from LoTAS 1 due to literally never being used:
- Right auto clicker
- Tick jump
- In-game timer
- Avoid taking damage (it was misused wayyyy too many times)
Mods that need to be rethought:
- All the options in the configuration menu
- The Info-HUD
- Entity Ai Manipulation
- Spawn manipulation
## After this
Once the GUI has been made maybe I'll find more motivation to add new mods and fix the old ones... Technically LoTAS can be released at that point already but I'm not sure that's a good idea.

**Comment by ScribbleTAS** [comment]
I'll try and get some mockups done... How are textures and icons looking, can they be integrated too?

**Comment by PancakeTAS** [comment]
yes

**Comment by ScribbleTAS** [comment]
4 Months ago, I made this mockup of the new LoTAS 3 Gui with the old Minecraft textures... We discussed that, when hitting a button in the bottom row, that the middle buttons should be cleared and the settings for e.g. Drop Manip should come up instead... I assume this won't be used anymore...
![PauseScreen2](https://github.com/MinecraftTAS/LoTAS/assets/28013575/78dd8dc3-6c10-4e13-8831-71633657a0cd)

I do think, that visually, the vanilla style buttons go together really well, but it is how it is...
And this is how the general layout is supposed to be?
![Pause Menu](https://github.com/MinecraftTAS/LoTAS/assets/28013575/c6c81e0d-412d-4975-82c4-ea11f43a2c59)


## Questions
### Main Menu
A lot of things are only available ingame, such as duping and savestates... Tickratechanger might be the only thing that is universal anywhere in MC, so does it make sense to support the main menu and all guis that are not ingame?

### Small window sizes
The gui gets really busy at some point on small window sizes... Something to keep in mind as well...

**Comment by PancakeTAS** [comment]
> And this is how the general layout is supposed to be?

Yes, this is how I had it in mind. Although I'm not sure of the right "quick tools" should be used for extra settings as well.

> so does it make sense to support the main menu and all guis that are not ingame?

Nah I just said that it's theoretically possible, but I don't think we should allow the menu to be opened in the main menu

> The gui gets really busy at some point on small window sizes... Something to keep in mind as well...

With LoTAS 3 I scaled it with the gui scale and some heavy floating point math. I ended up having stupid annoyances with 1% sometimes being 1 pixel vertically and 2 pixels horizontally, or text scaling weirdly. I don't really think we need to go through all that trouble just to support LoTAS when it's in it's 854x480 non-fullscreen window. Not like anyone uses that anyways...

**Comment by PancakeTAS** [comment]
Also I would like to add that I don't want to go for the minecraft-style of gui. I want something like I had previously with LoScreens and feel free to do it in Minecraft TAS blue instead of green.
